manager/src/main/java/org/openremote/manager/mqtt/TeltonikaMQTTHandler.java

```diff
@@ -0,0 +1,627 @@
+    }
+    @Override
+    public boolean checkCanSubscribe(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        // Skip standard checks
+        if (!canSubscribe(connection, securityContext, topic)) {
+            getLogger().warning("Cannot subscribe to this topic, topic=" + topic + ", connection" + connection);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if the Subscribing client should be allowed to subscribe to the topic that is handled by this Handler.
+     * For Teltonika device endpoints, we need the fourth token (Index 3) to be a valid IMEI number.
+     * We do that by checking using IMEIValidator.
+     */
+    // To be removed when auto-provisioning works
+    @Override
+    public boolean canSubscribe(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        if(topic.getTokens().size() < 5){
+            getLogger().warning(MessageFormat.format("Topic {0} is not a valid Topic. Please use a valid Topic.", topic.getString()));
+            return false;
+        }
+        long imeiValue;
+        try{
+            imeiValue = Long.parseLong(topic.getTokens().get(3));
+        }catch (NumberFormatException e){
+            getLogger().warning(MessageFormat.format("IMEI {0} is not a valid IMEI value. Please use a valid IMEI value.", topic.getTokens().get(3)));
+            return false;
+        }
+        //TODO: fix the boolean expression that is commented out below
+
+        if(!topic.getTokens().get(2).equalsIgnoreCase(TELTONIKA_DEVICE_TOKEN)) return false;
+        if(!IMEIValidator.isValidIMEI(imeiValue)) return false;
+        if(Objects.equals(topic.getTokens().get(4), TELTONIKA_DEVICE_RECEIVE_TOPIC)) return true;
+        if(Objects.equals(topic.getTokens().get(4), TELTONIKA_DEVICE_SEND_TOPIC)) return true;
+
+//        return true;
+
+        return Objects.equals(topic.getTokens().get(2), TELTONIKA_DEVICE_TOKEN) &&
+                IMEIValidator.isValidIMEI(imeiValue) &&
+                (
+                        Objects.equals(topic.getTokens().get(4), TELTONIKA_DEVICE_RECEIVE_TOPIC) ||
+                        Objects.equals(topic.getTokens().get(4), TELTONIKA_DEVICE_SEND_TOPIC)
+                );
+    }
+
+    /**
+     * Overrides MQTTHandler.checkCanPublish for this specific Handler,
+     * until secure Authentication and Auto-provisioning
+     * of Teltonika Devices is created.
+     * To be removed after implementation is complete.
+     */
+    @Override
+    public boolean checkCanPublish(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        return canPublish(connection,securityContext, topic);
+    }
+
+    @Override
+    public boolean canPublish(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        getLogger().info("Teltonika device will publish to Topic "+topic.toString()+" to transmit payload");
```

Comment:
Log level should be lower as this will be hit on every device publish

Reply:
set to `FINER`